### DISCUSSION TITLE
Oqtane 6.0 Released!

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#4833] [General]
The 6.0 release is a major release and represents the official migration from .NET 8 to .NET 9 which includes usability, security and performance improvements.

With the release of 6.0, Oqtane is fully transitioning its development efforts to .NET 9. And in alignment with our philosophy, there is a seamless and automated upgrade path from all prior Oqtane releases to the 6.0 release. In addition, Oqtane modules and themes that were created for prior versions will continue to function properly on 6.0 as the framework remains fully backward compatible.

To date, the open source project has recorded over 2685 pull requests from 57 contributors and has published 50 official releases, which ranks it among the most active open source projects within the .NET Foundation. The Oqtane framework continues to evolve at a rapid pace to meet the needs of the .NET community. 

https://github.com/oqtane/oqtane.framework/releases/tag/v6.0.0

**Comment by sbwalker** [comment]
For those of you hosting your Oqtane sites in Microsoft Azure...

You most likely installed a .NET 8 SDK patch version sometime in the past year using the Development Tools / Extensions to ensure that your Azure App Service was running the latest version of .NET 8. When you want to upgrade to Oqtane 6.0 on .NET 9 you will need to do 2 things:

1. Set the the .NET Version to .NET 9 (STS) in Settings / Configuration

![image](https://github.com/user-attachments/assets/b5e17d4f-7db8-46bd-af09-c264cd3df4ed)

2. Remove the .NET 8 SDK patch version from Development Tools / Extensions

![image](https://github.com/user-attachments/assets/3a4afbad-3720-4450-823b-479addaced68)

Then Restart your App Service.

If you do not follow these steps you will get a "HTTP Error 500.31 - Failed to load ASP.NET Core runtime" error when you browse to the site

Once you have upgraded you will see "Runtime Stack: Dotnet - v9.0" on the main Overview page:

![image](https://github.com/user-attachments/assets/160d2910-5463-49ab-b9d1-f6756708d52c)

And Oqtane reports in System Info

![image](https://github.com/user-attachments/assets/cb6a584f-6ee8-43e1-926b-91fc99ecc48d)

**Reply by sbwalker** [reply]
One other note related to Azure... I usually deploy Oqtane on a Business B1 plan... but I just realized for $9 more per month you can use Premium V3 P0V3 which provides way more CPU, RAM, and storage:

![PremiumV3](https://github.com/user-attachments/assets/75d2ea3e-5fc4-4308-acab-35e887eca9bb)

**Reply by mdmontesinos** [reply]
What OS and Region are those prices from? They're extremely cheap! I'm thinking about deploying Oqtane to Linux App Service, but I wasn't sure if it would work properly.

**Reply by sbwalker** [reply]
@mdmontesinos 

![image](https://github.com/user-attachments/assets/c1fe8ac5-93e1-41b8-884f-81e2c5e4e8fd)

**Reply by mdmontesinos** [reply]
This is the official pricing (https://azure.microsoft.com/en-us/pricing/details/app-service/windows/)

![image](https://github.com/user-attachments/assets/3664ef38-ec1c-4d29-a6ed-a2b79a4ec324)

In this case the difference between B1 and P0v3 is more than $50.

**Reply by sbwalker** [reply]
This Oqtane instance is powering 10 different websites. The Azure App Service has 10 custom domains defined and it is using the free Azure SSL certificates (ie. Digicert) for the domains (which renew automatically). Oqtane obviously requires a database and I am using a SQL Database with a Basic service tier which costs $5/mo. So my hosting bill for these websites is essentially $40 + $5 / 10 = $4.50/mo.

**Reply by sbwalker** [reply]
@mdmontesinos I guess my rates are lower because of my Subscription - I am not using Pay-As-You-Go
